### DISCUSSION TITLE
The reason why router cache only caches RSC Payload?

### DISCUSSION THREAD

**Discussion by algoORgoal** [vercel/next.js#65555] [Help]
### Summary

I have a question about the cacheing mechanism in Next.js
![image](https://github.com/vercel/next.js/assets/19303532/6c6a00db-9cee-4f51-afd7-dea2e3f5845d)
According to the docs, Router cache only caches RSC Payload, not the HTML. Meanwhile, full route cache caches RSC Payload as well as HTML. Is there any reason why the client-side router cache doesn't cache HTML as well? Would it be less performant to do such things?

Thank you.
Here is the part of the docs I'm referring to: https://nextjs.org/docs/app/building-your-application/caching#router-cache

**Comment by algoORgoal** [comment]
I have a theory that I just came up with, but I'm not sure if it's correct.
=> Since HTML is solely used to create RSC Payload, client side doesn't need to cache it. The full route cache caches HTML to generate RSC Payload faster during revalidation.

**Comment by nayounsang** [comment]
I'm curious about this too.
Next.js renders HTML **on server** using RSC payload and Client Component Javascript Instructions.
HTML is rendered not on client.
After client receives HTML to immediately show preview, RSC payload reconciles component trees and updates DOM on client.
To reduce server reqs on navigation, is caching HTML really unnecessary?
